rmw_connext_cpp/src/functions.cpp

```diff
@@ -1466,14 +1489,33 @@ rmw_destroy_service(rmw_service_t * service)
     service->implementation_identifier, rti_connext_identifier,
     return RMW_RET_ERROR)
 
-  auto result = RMW_RET_OK;
-  // TODO(esteve): de-allocate Replier and request DataReader
-  RMW_TRY_DESTRUCTOR(
-    static_cast<ConnextStaticServiceInfo *>(service->data)->~ConnextStaticServiceInfo(),
-    ConnextStaticServiceInfo,
-    result = RMW_RET_ERROR)
+  ConnextStaticServiceInfo * service_info = static_cast<ConnextStaticServiceInfo *>(service->data);
+
+  if (service_info) {
+    auto request_datareader = service_info->request_datareader_;
+    if (request_datareader) {
+      auto read_condition = service_info->read_condition_;
+      if (read_condition) {
+        if (request_datareader->delete_readcondition(read_condition) != DDS_RETCODE_OK) {
+          RMW_SET_ERROR_MSG("failed to delete readcondition");
+          return RMW_RET_ERROR;
```

Comment:
This should only set the return code but continue to perform the other clean up operations.

Same below and above.